distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java

```diff
@@ -88,6 +79,14 @@ static List<String> systemJvmOptions(Settings nodeSettings, final Map<String, St
         ).flatMap(s -> s).toList();
     }
 
+    private static String entitlementJarPropertyOption(String libraryName, Path workingDir) {
+        String jarSuffix = "-" + Build.current().version() + (Build.current().isSnapshot()? "-SNAPSHOT" : "") + ".jar";
```

Comment:
I still think that getting "help" from gradle here, and write the actual jar file name in the manifest or a resource, would be more robust, but this should do for now.
(AFAIK, SNAPSHOT is the only suffix we still care about, BETA, ALPHA etc. are retired - @rjernst ?)

Reply:
Ok I think I figured it out: I just need `project.version`.

Reply:
> Ok I wasn't sure of the most robust way to turn the build-time jar location into the run-time location.

I would look at how we do for native; a combination of loading from the working dir (check `LoaderHelper#findPlatformLibDir`) and system properties (e.g. `es.nativelibs.path`) for tests.

Reply:
Unfortunately the pattern for native won't work here. Those are native libs, and unversioned.

However, each of the 3 jars should be independent (no dependencies). I suggest we contain them all in one directory under lib (you can keep it under lib/tools/entitlement-agent for now, I'll move it up a level later). Then do a listing of the directory, and use prefixes to find each jar. There should be exactly 3 jars, and we should find each of the prefixes we expect.

Reply:
Finding the jars in a precise lib subdirectory, by prefix (or pattern, like `agent*.jar`), looks like a good option to me. Simple, still robust enough.